include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -666,8 +692,8 @@ __pattern_transform_scan_multi_group(_ExecutionPolicy&& __exec, _InRng&& __in_rn
     _NoAssign __no_assign_op;
     _NoOpFunctor __get_data_op;
 
-    return oneapi::dpl::__par_backend_hetero::__parallel_transform_scan(
-        ::std::forward<_ExecutionPolicy>(__exec), __in_rng.all_view(), __out_rng.all_view(), __binary_op, __init,
+    return __future(__parallel_transform_scan_multi_group(
```

Comment:
Is `__future ` created here just by `event`? I think it is not quite correct.
Why do we no return a future returned by `__parallel_transform_scan_multi_group`?

Reply:
There are two branches in this function that both return futures. The first branch for the single-work-group version of the algorithm returns a future with no value, since a value is not needed. This is the second branch which calls the multiple work-group version of the algorithm that returns a future with a value. Since the value is not needed in the general case for transform scan, we drop the value here and return a future with just the event.

We cannot modify the multiple work group version's future to not return a value because it used by other algorithms that directly call that version, such as `copy_if`.

Reply:
It seems that the general confusion here and [above](https://github.com/oneapi-src/oneDPL/pull/853/files#r1206662794) both stem from the fact that `__pattern_transform_scan_multi_group` is called in two ways:  

1. The simple case which does not need a return value called here
2. The general case where more is embedded into the provided functors, and a value is expected in the `__future` return (like `copy_if`).  

Perhaps it makes sense to have a two functions to match these two use cases:  (1) which which serves the simple case called here, drops the value from the future with a comment, sets up the local, between group and global scans, and otherwise just calls the (2) which returns a future with a value and has a different name.  

I don't have good names for these functions, brainstorming some:
1. `__parallel_transform_scan_multi_group`
2. `__parallel_transform_scan_`  +  `general_purpose`, `builder`, or `_custom`

Reply:
To be clear, this would add a layer to the new diagram between `__parallel_transform_scan_multi_group` and `__parallel_scan_submitter` which would be (2) above.  That function would be the entry point for `copy_if` and other patterns which need the custom scan functionality.

Reply:
After discussing with @rarutyun, we decided to rename `__parallel_transform_scan_multi_group` to `__parallel_transform_scan_base` to make it clearer that this function is a generic version that supports multiple algorithms.

Reply:
Sounds fine to me.  I still think that separating into two functions (one which wraps the other) may be useful in the future, as it can clarify these two uses and their associated differences.